src/ConfigFile.c

```diff
@@ -678,6 +678,45 @@ void Ros_ConfigFile_ValidateNonCriticalSettings()
     }
 }
 
+void Ros_ConfigFile_PrintActiveConfiguration()
+{
+    Ros_Debug_BroadcastMsg("Config: ros_domain_id = %d", g_nodeConfigSettings.ros_domain_id);
+    Ros_Debug_BroadcastMsg("Config: node_name = '%s'", g_nodeConfigSettings.node_name);
+    Ros_Debug_BroadcastMsg("Config: node_namespace = '%s'", g_nodeConfigSettings.node_namespace);
+    Ros_Debug_BroadcastMsg("Config: remap_rules = '%s'", g_nodeConfigSettings.remap_rules);
+    Ros_Debug_BroadcastMsg("Config: agent_ip_address = '%s'", g_nodeConfigSettings.agent_ip_address);
+    Ros_Debug_BroadcastMsg("Config: agent_port_number = '%s'", g_nodeConfigSettings.agent_port_number);
+    Ros_Debug_BroadcastMsg("Config: sync_timeclock_with_agent = %d", g_nodeConfigSettings.sync_timeclock_with_agent);
+    Ros_Debug_BroadcastMsg("Config: namespace_tf = %d", g_nodeConfigSettings.namespace_tf);
+    Ros_Debug_BroadcastMsg("Config: publish_tf = %d", g_nodeConfigSettings.publish_tf);
+    Ros_Debug_BroadcastMsg("List of configured joint names:");
+    for (int i = 0; i < MAX_CONTROLLABLE_GROUPS; i += 1)
+    {
+        Ros_Debug_BroadcastMsg("---");
+        for (int j = 0; j < MP_GRP_AXES_NUM; j += 1)
+        {
+            if (strlen(g_nodeConfigSettings.joint_names[(i * MP_GRP_AXES_NUM) + j]) > 0)
+                Ros_Debug_BroadcastMsg(g_nodeConfigSettings.joint_names[(i * MP_GRP_AXES_NUM) + j]);
+            else
+                Ros_Debug_BroadcastMsg("x");
+        }
+    }
+    Ros_Debug_BroadcastMsg("---");
+    Ros_Debug_BroadcastMsg("Config: logging.log_to_stdout = %d", g_nodeConfigSettings.log_to_stdout);
+    Ros_Debug_BroadcastMsg("Config: update_periods.executor_sleep_period = %d", g_nodeConfigSettings.executor_sleep_period);
+    Ros_Debug_BroadcastMsg("Config: update_periods.action_feedback_publisher_period = %d", g_nodeConfigSettings.action_feedback_publisher_period);
+    Ros_Debug_BroadcastMsg("Config: update_periods.controller_status_monitor_period = %d", g_nodeConfigSettings.controller_status_monitor_period);
+    Ros_Debug_BroadcastMsg("Config: publisher_qos.robot_status = %d", g_nodeConfigSettings.qos_robot_status);
+    Ros_Debug_BroadcastMsg("Config: publisher_qos.joint_states = %d", g_nodeConfigSettings.qos_joint_states);
+    Ros_Debug_BroadcastMsg("Config: publisher_qos.tf = %d", g_nodeConfigSettings.qos_tf);
```

Comment:
minor, but these lines will now only print a nr (ie: the value of the `enum` member). They used to print the name of the QoS profile.

It would be nice if we could add a mapping from the `enum` integer to their string description again, as that would be much more human readable, but I won't block merging this PR for that (seeing as this was already accepted as part of #81.